crates/sqlexec/src/planner/physical_plan/remote_insert.rs

```diff
@@ -0,0 +1,128 @@
+use core::fmt;
+use std::sync::Arc;
+
+use datafusion::{
+    arrow::datatypes::{DataType, Field, Schema},
+    error::DataFusionError,
+    execution::context::SessionState,
+    physical_plan::{
+        coalesce_partitions::CoalescePartitionsExec, stream::RecordBatchStreamAdapter, DisplayAs,
+        DisplayFormatType, ExecutionPlan,
+    },
+};
+use futures::{stream, TryStreamExt};
+use once_cell::sync::Lazy;
+use uuid::Uuid;
+
+use super::{client_recv::ClientExchangeRecvExec, remote_scan::ProviderReference};
+
+pub static INSERT_PLAN_SCHEMA: Lazy<Arc<Schema>> = Lazy::new(|| {
+    Arc::new(Schema::new(vec![Field::new(
+        "count",
+        DataType::UInt64,
+        /* nullable = */ false,
+    )]))
+});
+
+/// Executes an insert on the cached table provider on remote side.
+///
+/// Only valid to execute this on remote side.
+#[derive(Debug, Clone)]
+pub struct RemoteInsertExec {
+    pub provider: ProviderReference,
+    pub input: Arc<ClientExchangeRecvExec>,
```

Comment:
This should be `Arc<dyn ExecutionPlan>`. Since it's executing on the remote side, it'll already have the plan ready to go and we don't need to do any client recv/send stuff.